lib/chef/mixin/openssl_helper.rb

```diff
@@ -401,6 +401,30 @@ def renew_x509_crl(crl, ca_private_key, info)
         crl.sign(ca_private_key, ::OpenSSL::Digest::SHA256.new)
         crl
       end
+
+      # Return true if a certificate need to be renewed (or doesn't exist) according to the number
+      # of days before expiration given
+      # @param [string] cert_file path of the cert file or cert content
+      # @param [integer] renew_before_expiry number of days before expiration
+      # @return [true, false]
+      def cert_need_renewall?(cert_file, renew_before_expiry)
+        raise TypeError, "cert_file must be a String object" unless cert_file.is_a?(String)
+        raise TypeError, "renew_before_expiry must be a Integer object" unless renew_before_expiry.is_a?(Integer)
```

Comment:
No need for this check if the property is an Int. Chef will blow up before it gets here.

Reply:
True. I removed it.